src/ralph/assets/subscribers.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+import logging
+import pyhermes
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+
+from ralph.assets.models.assets import Environment, Service, ServiceEnvironment
+
+logger = logging.getLogger(__name__)
+
+
+def _update_service_owners(service, business_owners, technical_owners):
+    """
+    Update service owners
+
+    Args:
+        service: Service
+        business_owners: dict of owners
+        technical_owners: dict of owners
+    """
+    business_owners = [
+        get_user_model().objects.get_or_create(username=i['username'])[0]
+        for i in business_owners
+    ]
+    technical_owners = [
+        get_user_model().objects.get_or_create(username=i['username'])[0]
+        for i in technical_owners
+    ]
+
+    service.business_owners = business_owners
+    service.technical_owners = technical_owners
+
+
+def _update_service_environments(service, environments):
+    """
+    Update service environments.
+
+    Args:
+        service: Service
+        environments: list of environment names
+    """
+    new_envs = []
+    for env_name in environments:
+        new_envs.append(Environment.objects.get_or_create(name=env_name)[0])
+
+    current = set([u.id for u in service.environments.all()])
+    new = set([u.id for u in new_envs])
+    to_delete = current - new
+    to_add = new - current
+    for env_id in to_add:
+        ServiceEnvironment.objects.create(
+            service=service,
+            environment_id=env_id
+        )
+    for env_id in to_delete:
+        ServiceEnvironment.objects.get(
+            service=service, environment_id=env_id
+        ).delete()
+
+
+@pyhermes.subscriber(topic=settings.HERMES_SERVICE_TOPICS['CREATE'])
+@pyhermes.subscriber(topic=settings.HERMES_SERVICE_TOPICS['UPDATE'])
+@pyhermes.subscriber(topic=settings.HERMES_SERVICE_TOPICS['REFRESH'])
+def update_service_handler(data):
```

Comment:
I'd use `data` name only as a last resort (i.e. when you **really** can't come up with anything more descriptive).